schema-nft-marketplace.graphql

```diff
@@ -0,0 +1,255 @@
+    marketplaceRevenueETH: BigDecimal!
+
+    " Revenue that goes to creator, aka royalty fee. "
+    creatorRevenueETH: BigDecimal!
+
+    " Sum of marketplaceRevenue and creatorRevenue. "
+    totalRevenueETH: BigDecimal!
+
+    " Trade count of the collection on the marketplace. "
+    tradeCount: Int!
+
+    " Trades of the collection. "
+    trades: [Trade!]! @derivedFrom(field: "collection")
+}
+
+" Trades exist such as a combination of taker/order and bid/ask. "
+type Trade @entity {
+    " { maker txn hash }-{ taker txn hash } "
+    id: ID!
+
+    " Transaction hash of the maker. "
+    makerTransactionHash: String!
+
+    " Transaction hash of the taker. "
+    takerTransactionHash: String!
+
+    " Event log index. "
+    logIndex: Int!
+
+    " Block timestamp where the trade is executed. "
+    timestamp: BigInt!
+
+    " Block number where the trade is executed. "
+    blockNumber: BigInt!
+
+    " Collection involved "
+    collection: Collection!
+
+    " Token ID of the traded NFT. "
+    tokenId: BigInt!
+
+    " Price (in ETH) "
+    priceETH: BigDecimal!
+
+    " Stretegy that the trade is executed. "
+    strategy: SaleStrategy!
+
+    " Buyer account address "
+    buyer: String!
+
+    " Seller account address "
+    seller: String!
+}
+
+type MarketplaceDailySnapshot @entity {
+    " { Contract address }-{# of days since Unix epoch time} "
+    id: ID!
+
+    " Block number where the snapshot is taken. "
+    blockNumber: BigInt!
+
+    " Block timestamp when the snapshot is taken. "
+    timestamp: BigInt!
+
+    " Number of collections listed on the marketplace. "
+    collectionCount: Int!
+
+    " Cumulative trade volume (in ETH) "
+    cumulativeTradeVolumeETH: BigDecimal!
+
+    " Revenue that goes to the marketplace protocol, aka protocol fee. "
+    marketplaceRevenueETH: BigDecimal!
+
+    " Revenue that goes to creator, aka royalty fee. "
+    creatorRevenueETH: BigDecimal!
+
+    " Sum of marketplaceRevenueETH and creatorRevenueETH. "
+    totalRevenueETH: BigDecimal!
+
+    " Trade count of the all collections on the marketplace. "
+    tradeCount: Int!
+
+    " Cumulative unique traders "
+    cumulativeUniqueTraders: Int!
+
+    " Daily active traders "
+    dailyActiveTraders: Int!
+
+    " Number of traded collections of the day "
+    dailyTradedCollectionCount: Int!
+
+    " Number of traded items of the day "
+    dailyTradedItemCount: Int!
+}
+
+type CollectionDailySnapshot @entity {
+    " { Contract address }-{# of days since epoch unix time } "
+    id: ID!
+
+    " Block number where the snapshot is taken. "
+    blockNumber: BigInt!
+
+    " Block timestamp when the snapshot is taken. "
+    timestamp: BigInt!
+
+    " Royalty fee rate in percentage. E.g. 2.5% should be 2.5 "
+    royaltyFee: BigDecimal!
+
+    " Number of tokens listed for sale in the marketplace. "
+    listedCount: BigInt!
+
+    " Floor price (in ETH) "
+    floorPriceETH: BigDecimal!
```

Comment:
why are we not interested in storing prices in USD?

Reply:
Because it's kinda a pain. We either need to calculate USD prices which will slow down indexing by a lot, or we need to store each currency denomination separately, which complicates the schema. 99.9% of the stuff are denominated in ETH anyways, so I think it's fine using ETH and omitting NFTs that are prices in other tokens.

Reply:
Normally, a marketplace uses at max 3 different tokens for trading, I don't think it will slow down indexing that much. btw are we only targeting `mainnet` marketplaces? If not, NFT marketplaces on different chains are not dependent on ETH.

Reply:
since our first phase involves mainly: opensea, looksrare and x2y2, they are all majorly mainnet marketplace so we are good :) we can always update the schema when needed

Reply:
@0xbe1 do you think we should drop the ETH, this was it is more generic and can apply to other chains in the future? it could look like:

```
" Floor price (in native asset) "
floorPrice: BigInt!
```
If we did it like this we could have a field in the `MarketPlace` called `nativeToken` that contains a type like `Token` in other subgraphs.. Food for thought

Reply:
Yeah, I think this is a good idea. But I guess it should still be bigDecimal

Reply:
@dmelotik good idea!

though i prefer going by ETH in this iteration for simplicity, as there are not many marketplace that matters living outside mainnet